python3/httplib2/__init__.py

```diff
@@ -181,14 +181,15 @@ def safename(filename):
         pass
     if isinstance(filename,str):
         filename=filename.encode('utf-8')
-    filemd5 = _md5(filename).hexdigest().encode('utf-8')
+    filemd5 = _md5(filename).hexdigest()
     filename = re_url_scheme.sub(b"", filename)
-    filename = re_slash.sub(b",", filename)
+    #filename = re_slash.sub(b",", filename)
+    filename = urllib.parse.quote(filename, safe='')
 
     # limit length of filename
-    if len(filename)>200:
-        filename=filename[:200]
-    return b",".join((filename, filemd5)).decode('utf-8')
+    if len(filename) > 90:
```

Comment:
> Now it is a maximum filename length, which is 259 in Windows.   
```
C:\Users\    <username>    \AppData\Local\Temp\  <safe_filename>  ,   <md5>
  9 chars + max 104 chars  +     20 chars      +       x       +  1  +  32  = max 259 chars
```

* please, add comment with reason and formula from above
* IMHO, unconditional trim is better, use your judgement

Reply:
That's right, unconditional trim is exactly that I'm suggesting, I just calculated the safe limit of the unconditional trim so that it would work under any circumstances.